src/ai/backend/agent/kernel.py

```diff
@@ -233,6 +235,9 @@ def __getstate__(self) -> Mapping[str, Any]:
         return props
 
     def __setstate__(self, props) -> None:
+        # Used when a `Kernel` object is loaded from pickle data.
+        if "state" not in props:
+            props["state"] = KernelLifecycleStatus.RUNNING
```

Comment:
When we shutdown an agent, `kernel_registry` is dumped as a pickle file and the agent loads the pickle file when it restarts.
When we update an agent and restart it, old `Kernel` objects that are dumped before update do not have `state` field.
We need to insert the `state` value to the old `Kernel` objects.